### ISSUE TITLE
New warning in the packaging linux jobs

### ISSUE THREAD

**Issue by Blast545** [ros2/rcutils#352]
Quick discussion issue, to get some thoughts. I see this new warning in the packaging jobs:
https://ci.ros2.org/view/packaging/job/packaging_linux/2565/gcc/new/folder.-1643903456/category.1304701477/

```
`‘strnlen’ specified bound 1024 exceeds source size 768 [-Wstringop-overread] 57 | size_t src_length = strnlen(src, dst_size); | ^~~~~~~~~~~~~~~~~~~~~~`
```

It comes from here: https://github.com/ros2/rcutils/blob/befc608e21c653404f2fd876986969286125a527/src/error_handling_helpers.h#L57

AFAIK, `strenlen` is expected to take "at most" `dst_size` bytes from source, and this warning seem to be expecting `dst_size` to be at most `strenlen(src)` which would defeat the purpose of `strnlen` with a second parameter.

Any ideas about this one @clalancette?

**Comment by Blast545** [comment]
I can try it out! I'll open a draft PR and check

**Comment by Blast545** [comment]
Closed with #353 

Reference newer case without the warning:
https://ci.ros2.org/view/packaging/job/packaging_linux/2572/
